system/Model.php

```diff
@@ -908,7 +912,7 @@ public function delete($id = null, bool $purge = false)
 
 		if ($this->useSoftDeletes && ! $purge)
 		{
-			$set[$this->deletedField] = 1;
+			$set[$this->deletedField] = $this->setDate();
```

Comment:
This should be wrapped inside of a `if ($this->useTimestamps)` similar to the `updatedField` just below. You can wrap them together if you need to.

Reply:
I think we need to set `deleted_at` regardless of whether `useTimestamps` is toggled. Though that brings up an interesting scenario of what to do if `useSoftDeletes` is true and `useTimestamps` is false, since `setDate` relies on `dateFormat` which is tied to `useTimestamps`.

Reply:
Good point. Deleting shouldn't be dependent on that, so ignore my comment here. :) 

I think the times when `useSoftDeletes` is true, `useTimestamps` is false, and `dateFormat` is empty should be extremely rare, since `dateFormat` has a default in the Model. It wouldn't hurt to check for that condition and throw an exception with a explantation that `dateFormat` cannot be empty.

Reply:
Thinking about this during dinner, I believe that `dateFormat` should be a requirement (throw if missing) for `useSoftDeletes` and `useTimestamps` but those two should not otherwise be related - that helps keep it clean and only check what is absolutely necessary. If you agree I'll resolve this convo and plan those sanity checks for a separate PR.

Reply:
Absolutely agree with that. Sounds good.